tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -10913,6 +10913,22 @@ create_global_initializer(GBL_LIST *gitem, const char *flag_str,
   gitem->global_def = gname;
 }
 
+/**
+   \brief Check if sptr is the midnum of a scalar and scalar has POINTER/ALLOCATABLE attribute
+   \param sptr  A symbol
+ */
+bool
+pointer_scalar_need_debug_info(SPTR sptr)
+{
+  if ((sptr > NOSYM) && REVMIDLNKG(sptr)) {
+    SPTR scalar_sptr = (SPTR)REVMIDLNKG(sptr);
+    if ((POINTERG(scalar_sptr) || ALLOCATTRG(scalar_sptr)) &&
+        ((STYPEG(scalar_sptr) == ST_VAR) || (STYPEG(scalar_sptr) == ST_STRUCT)))
```

Comment:
When will `STYPEG(scalar_sptr)` be `ST_STRUCT` for a deferred-length string pointer?

Reply:
Hi, as shared earlier or below, this PR contains fix for both deferred string and scalar var debug support, hence the function added (pointer_scalar_need_debug_info) is part of scalar pointer debug info support. Also there was some inter dependency b/w deferred string and scalar pointer support, hence i had to club both in this PR. any other details i can share in detail, please let me know, thanks for the comments

>This fix also emit debug info for scalar pointer as shown below. without this fix, scalar pointer variables can not be debugged or printed with GDB. Now this fix emits the required debug info for these scalar pointer variables and we can print scalar pointer variables inside GDB. Hence this PR address both deferred length string and scalar pointer debug info support.

program pointerExample
implicit none
integer, pointer :: p1
end program pointerExample

Reply:
with this PR, we will enable deferred string and scalar var debug support